src/metrics.cpp

```diff
@@ -121,8 +153,43 @@ int printMetrics(size_t cols, int64_t nStart, bool mining)
 
         int mined = minedBlocks.get();
         if (mined > 0) {
+            LOCK(cs_main);
+            boost::strict_lock_ptr<std::list<uint256>> u = trackedBlocks.synchronize();
+            auto consensusParams = Params().GetConsensus();
+            auto tipHeight = chainActive.Height();
+            std::string units = Params().CurrencyUnits();
+
+            // Update orphans and calculate subsidies
+            CAmount immature {0};
+            CAmount mature {0};
+            for (std::list<uint256>::iterator it = u->begin(); it != u->end(); it++) {
+                auto hash = *it;
+                if (mapBlockIndex.count(hash) > 0 &&
+                        chainActive.Contains(mapBlockIndex[hash])) {
+                    int height = mapBlockIndex[hash]->nHeight;
+                    CAmount subsidy = GetBlockSubsidy(height, consensusParams);
+                    if ((height > 0) && (height <= consensusParams.GetLastFoundersRewardBlockHeight())) {
+                        subsidy -= subsidy/5;
+                    }
+                    if (std::max(0, COINBASE_MATURITY - (tipHeight - height)) > 0) {
+                        immature += subsidy;
+                    } else {
+                        mature += subsidy;
+                    }
+                } else {
+                    it = u->erase(it);
```

Comment:
The implicit assumption here is that if a block hash disappears either from `mapBlockIndex` (should only happen on restart, so never relevant here) or `chainActive` (happens on reorgs), it will never re-appear. I think this is a fair assumption for reorgs, inasmuch as it is far less likely that a second reorg would occur after the local node has already observed one. But if we want to remove that assumption, I can just build two parallel lists (one for tracked blocks, one for tracked orphans) and shuttle block hashes between them.